react/features/lobby/components/AbstractLobbyScreen.js

```diff
@@ -0,0 +1,317 @@
+
+    /**
+     * Function to be used to translate i18n labels.
+     */
+    t: Function
+};
+
+type State = {
+
+    /**
+     * The display name value entered into the field.
+     */
+    displayName: string,
+
+    /**
+     * The email value entered into the field.
+     */
+    email: string,
+
+    /**
+     * The password value entered into the field.
+     */
+    password: string,
+
+    /**
+     * The state of the screen. One of {@code SCREEN_STATES[*]}
+     */
+    screenState: number
+}
+
+/**
+ * Abstract class to encapsulate the platform common code of the {@code LobbyScreen}.
+ */
+export default class AbstractLobbyScreen extends PureComponent<Props, State> {
+    /**
+     * Instantiates a new component.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            displayName: props._participantName || '',
+            email: props._participantEmail || '',
+            password: '',
+            screenState: props._participantName ? SCREEN_STATES.VIEW : SCREEN_STATES.EDIT
+        };
+
+        this._onAskToJoin = this._onAskToJoin.bind(this);
+        this._onCancel = this._onCancel.bind(this);
+        this._onChangeDisplayName = this._onChangeDisplayName.bind(this);
+        this._onChangeEmail = this._onChangeEmail.bind(this);
+        this._onChangePassword = this._onChangePassword.bind(this);
+        this._onEnableEdit = this._onEnableEdit.bind(this);
+        this._onSwitchToKnockMode = this._onSwitchToKnockMode.bind(this);
+        this._onSwitchToPasswordMode = this._onSwitchToPasswordMode.bind(this);
+    }
+
+    /**
+     * Returns the screen title.
+     *
+     * @returns {string}
+     */
+    _getScreenTitleKey() {
+        const withPassword = Boolean(this.state.password);
+
+        return this.props._knocking
+            ? withPassword ? 'lobby.joiningWithPasswordTitle' : 'lobby.joiningTitle'
+            : 'lobby.joinTitle';
+    }
+
+    _onAskToJoin: () => void;
+
+    /**
+     * Callback to be invoked when the user submits the joining request.
+     *
+     * @returns {void}
+     */
+    _onAskToJoin() {
+        this.props.dispatch(startKnocking(this.state.password));
+
+        return false;
+    }
+
+    _onCancel: () => boolean;
+
+    /**
+     * Callback to be invoked when the user cancels the dialog.
+     *
+     * @private
+     * @returns {boolean}
+     */
+    _onCancel() {
+        this.props.dispatch(cancelKnocking());
+
+        return true;
+    }
+
+    _onChangeDisplayName: Object => void;
+
+    /**
+     * Callback to be invoked when the user changes its display name.
+     *
+     * @param {SyntheticEvent} event - The SyntheticEvent instance of the change.
+     * @returns {void}
+     */
+    _onChangeDisplayName(event) {
+        const displayName = getFieldValue(event);
+
+        this.setState({
+            displayName
+        }, () => {
+            this.props.dispatch(updateSettings({
```

Comment:
given that we are not relying on props for handling ssubsequent state changes, does it matter to update settings right after calling setState, aka, before it may have completed?

Reply:
Not really, but it doesn't hurst I guess